sound/soc/intel/boards/sof_pcm512x.c

```diff
@@ -360,6 +360,10 @@ static int sof_audio_probe(struct platform_device *pdev)
 	} else {
 		dmic_be_num = 2;
 #if IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI)
+		if (!mach->mach_params.common_hdmi_codec_drv) {
+			dev_err(&pdev->dev, "HDMI codec driver error\n");
+			return -EINVAL;
+		}
```

Comment:
@kv2019i I might be wrong, but doesn't this kill configurations, where we want to boot with a generic kernel, where both HDMI codecs are enabled, but where for that specific platform we don't want or cannot yet use the common HDMI driver?